internal/snykclient/sbomconvert.go

```diff
@@ -33,25 +33,32 @@ func (t *SnykClient) SBOMConvert(
 		sbom,
 	)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create request: %w", err)
+		return nil, errFactory.NewSCAError(err)
 	}
 
 	req.Header.Set(ContentTypeHeader, MIMETypeOctetStream)
 
 	resp, err := t.client.Do(req)
 	if err != nil {
-		return nil, fmt.Errorf("failed to send request: %w", err)
+		return nil, errFactory.NewSCAError(err)
 	}
 	defer resp.Body.Close()
 
-	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
-		return nil, fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+	// TODO: when server-side things go wrong, we should be displaying a
+	// request ID, interaction ID.
+
+	if resp.StatusCode > 399 && resp.StatusCode < 500 {
+		return nil, errFactory.NewSCAError(fmt.Errorf("request to analyze SBOM document was rejected: %s", resp.Status))
+	}
+
+	if resp.StatusCode > 499 {
+		return nil, errFactory.NewSCAError(fmt.Errorf("analysis of SBOM document failed due to error: %s", resp.Status))
```

Comment:
I think we are good to merge this into the happy-path build, but I suggest the 2 following steps:
1. Review the message ourselves with UX's team's instructions: https://snyksec.atlassian.net/wiki/spaces/PD/pages/1819508891/Writing+error+messages
2. Ask UX team for review

In this error message, for example, just from reading the code I would expect output similar to:
> There was an error while analyzing the SBOM document: analysis of SBOM document failed due to error: Internal Error

We could provide more guidance on what user can do to fix the problem - for example, ask to try again or try later and contact support if the issue persists.

Reply:
absolutely! in fact, there's an `error-catalog` package we could use which renders an even nicer ux experience. i'd regard these as placeholders either _before_ the closed beta or during the closed beta.